src/engine/Match_tainting_mode.ml

```diff
@@ -472,56 +474,141 @@ let check_rule per_file_formula_cache (rule : R.taint_rule) match_hook
               ([], Shape_and_sig.FunctionMap.empty)
               ast
           in
-          let collected_infos = List.rev collected_infos in
+          (* Use object mappings from Object_initialization.ml *)
+          let all_object_mappings = object_mappings in
+          let initial_signature_db =
+            Shape_and_sig.add_object_mappings base_db all_object_mappings
+          in
 
+          (* Use shared call graph if provided, otherwise compute it *)
           let call_graph =
-            Function_call_graph.build_call_graph ~lang ~object_mappings ast
+            match shared_call_graph with
+            | Some (graph, shared_mappings) ->
+                (* Verify that object_mappings match to ensure consistency *)
+                if
+                  List.length shared_mappings
+                  <> List.length all_object_mappings
```

Comment:
or make the match case :

```
| ..... when Int.equal (List.length shared_mappings) (List.length all_object_mappings)
```

But why could this happen? 
Under which conditions?